### PR TITLE
[TurbConv] Fold state vec funcs

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR does two things:

1) Add functions for the state
 - surface_val
 - first_elem_above_surface_val
 - domain_average!
 - distribute!
 - total_covariance!
 - extrap!
 - assign_ghost!
 - integrate_ode!
 - export_state
 - plot_state

2) Renames `Slice` in `StateVec` to `Cut`, to avoid overloading issues (which I experienced with tests in the PR following this one).

### CODE REVIEW

File: src/ClimaAtmos/Parameterizations/TurbulenceConvection/test/StateVecFuncs.jl [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  integrate_ode!(state_vec, :w, grid, func, c_0, NamedTuple(), 1)
+  sol_approx = [state_vec[:w, k] for k in over_elems_real(grid)]
+  sol_percent_error = [(sol_approx[k]-sol_correct[k])/sol_correct[k]*100 for k in 1:length(sol_correct)]
+  @test all(sol_percent_error .< 100*grid.dz)
+end
+
+@static if haskey(Pkg.installed(), "Plots")
+  @testset "Plot state vector" begin
+    @test try
+      plot_state(state_vec, grid, "./", "a", :a)
```
Comment (simonbyrne) [inline src/ClimaAtmos/Parameterizations/TurbulenceConvection/test/StateVecFuncs.jl]: I'm not a huge fan of putting plots in tests. Can you generate a few sample values and check them?

File: src/ClimaAtmos/Parameterizations/TurbulenceConvection/src/StateVecs/StateVecFuncs.jl [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+                   name::Symbol,
+                   grid::Grid,
+                   func::Function,
+                   y_0::AbstractFloat,
+                   args::NamedTuple,
+                   i_sd::Int = 1)
+
+Integrates the ODE:
+  `dy/dz = func(z, args)`
+  `y = y_0 + int_{z=0}^{z} func(z, args) dz`
```
Comment (simonbyrne) [inline src/ClimaAtmos/Parameterizations/TurbulenceConvection/src/StateVecs/StateVecFuncs.jl]: A bit more documentation would be better (what does it return? what are the arguments?)
Note that you can do a block math quote by "```math ... ```"

File: src/ClimaAtmos/Parameterizations/TurbulenceConvection/src/StateVecs/StateVecFuncs.jl [hunk 3]
```diff
@@ -0,0 +1,322 @@
+"""
+    StateVecFuncs
+
+  Provides a set of helper functions that operate on StateVec.
+"""
+module StateVecFuncs
+
+using Pkg
+@static if haskey(Pkg.installed(), "Plots")
```
Comment (simonbyrne) [inline src/ClimaAtmos/Parameterizations/TurbulenceConvection/src/StateVecs/StateVecFuncs.jl]: use `Requires` for this.

File: src/ClimaAtmos/Parameterizations/TurbulenceConvection/src/StateVecs/StateVecFuncs.jl [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+A singleton used to indicate to use a .vtk
+extension when exporting a `StateVec`.
+"""
+struct UseVTK <: ExportType end
+
+"""
+    UseDat
+
+A singleton used to indicate to use a .dat
+extension when exporting a `StateVec`.
```
Comment (simonbyrne) [inline src/ClimaAtmos/Parameterizations/TurbulenceConvection/src/StateVecs/StateVecFuncs.jl]: I don't think `.dat` is a standard file format, maybe explain what it outputs.

File: src/ClimaAtmos/Parameterizations/TurbulenceConvection/src/StateVecs/StateVecFuncs.jl (hunk 2 above)
Comment (charleskawczynski) [reply src/ClimaAtmos/Parameterizations/TurbulenceConvection/src/StateVecs/StateVecFuncs.jl]: I knew that I left something that needed LaTeXification!

File: src/ClimaAtmos/Parameterizations/TurbulenceConvection/src/StateVecs/StateVecFuncs.jl (hunk 4 above)
Comment (charleskawczynski) [reply src/ClimaAtmos/Parameterizations/TurbulenceConvection/src/StateVecs/StateVecFuncs.jl]: Added a bit more info, let me know if you want more. I'd be fine changing this to `.csv` if that's easier.

**Pull request merged**